src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -358,7 +358,11 @@ private IMethodNode CreateMethodEntrypoint(TypeAndMethod key)
             MethodWithToken method = key.Method;
             bool isInstantiatingStub = key.IsInstantiatingStub;
             bool isPrecodeImportRequired = key.IsPrecodeImportRequired;
-            MethodDesc compilableMethod = method.Method.GetCanonMethodTarget(CanonicalFormKind.Specific);
+            MethodDesc compilableMethod = method.Method;
+            if (!compilableMethod.OwningType.IsArray)
```

Comment:
I wouldn't expect special casing outside corinfoimpl to be needed - RyuJIT should realize this is an array constructor and ask for a helper - I wouldn't expect there would be an entrypoint for the actual method.

Reply:
Thanks Michal for your insightful feedback. I have tracked this down to the fact that we need to set the <code>codePointerOrStubLookup.constLookup</code> in the "zapinfo" version of getCallInfo, otherwise it unconditionally requested the entrypoint. As of the 2nd commit I've been able to roll this back.